encord/orm/dataset.py

```diff
@@ -54,19 +58,93 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+@dataclasses.dataclass(frozen=True)
+class SignedUrl:
+    signed_url: str
+
+
+@dataclasses.dataclass(frozen=True)
+class ImageData:
+    id: int
+    title: str
+    user_hash: UUID
+    file_link: str
+    file_type: str
+    file_size: float
+    image_hash: UUID
+    storage_location: StorageLocation
+    created_at: datetime
+    last_edited_at: datetime
+    height: int
+    width: int
+
+    @classmethod
+    def from_dict(cls, json_dict: Dict):
+        return ImageData(
+            json_dict["id"],
+            json_dict["user_hash"],
+            json_dict["title"],
+            json_dict["file_link"],
+            json_dict["file_type"],
+            json_dict["file_size"],
+            json_dict["image_hash"],
+            json_dict["storage_location"],
+            parser.parse(json_dict["created_at"]),
+            parser.parse(json_dict["last_edited_at"]),
+            json_dict["height"],
+            json_dict["width"],
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class ImagesInGroup:
+    images: List[Dict]
```

Comment:
all complex data we get from BE are dicts. And we can convert them to specific types after receiving from the BE